packages/fast-tooling-react/src/form/form.tsx

```diff
@@ -172,54 +175,134 @@ class Form extends React.Component<
+                return new StandardControlPlugin({
+                    context: ControlContext.fill,
+                    type: ControlType.array,
+                    control: (config: ArrayControlConfig): React.ReactNode => {
+                        return <ArrayControl {...config} />;
+                    },
+                });
+            case ControlType.children:
+                return new StandardControlPlugin({
+                    context: ControlContext.fill,
+                    type: ControlType.children,
+                    control: (config: ChildrenControlConfig): React.ReactNode => {
+                        return <ChildrenControl {...config} />;
+                    },
+                });
+            case ControlType.numberField:
+                return new StandardControlPlugin({
+                    type: ControlType.numberField,
+                    control: (config: NumberFieldTypeControlConfig): React.ReactNode => {
+                        return <NumberFieldControl {...config} />;
+                    },
+                });
+            case ControlType.checkbox:
+                return new SingleLineControlPlugin({
+                    type: ControlType.checkbox,
+                    control: (config: CommonControlConfig): React.ReactNode => {
+                        return <CheckboxControl {...config} />;
+                    },
+                });
+            case ControlType.sectionLink:
+                return new StandardControlPlugin({
+                    type: ControlType.sectionLink,
+                    control: (config: SectionLinkControlConfig): React.ReactNode => {
+                        return <SectionLinkControl {...config} />;
+                    },
+                });
+            case ControlType.textarea:
+                return new StandardControlPlugin({
+                    type: ControlType.textarea,
+                    control: (config: TextareaControlConfig): React.ReactNode => {
+                        return <TextareaControl {...config} />;
+                    },
+                });
+            case ControlType.display:
+                return new StandardControlPlugin({
+                    type: ControlType.display,
+                    control: (config: ControlConfig): React.ReactNode => {
+                        return <DisplayControl {...config} />;
+                    },
+                });
+            case ControlType.button:
+                return new StandardControlPlugin({
+                    type: ControlType.button,
+                    control: (config: ControlConfig): React.ReactNode => {
+                        return <ButtonControl {...config} />;
+                    },
+                });
+        }
+    }
+
+    private updateControls(): void {
+        const hasCustomControlPlugins: boolean = Array.isArray(this.props.controlPlugins);
+
+        this.selectControl = this.findControlPlugin(
+            hasCustomControlPlugins,
```

Comment:
If the logic for determining if there are custom control plugins is always the same, is should `this.findControlPlugin` just determine this itself? Then you don't have to pass it as an argument for every invocation.

Reply:
Passing it as an argument means I only have to do the check once, otherwise I would do the same conditional 9 times.